server/server.go

```diff
@@ -616,6 +624,18 @@ func NewServer(userConfig UserConfig, config Config) (*Server, error) {
 		DB:                 boltdb,
 		DeleteLockCommand:  deleteLockCommand,
 	}
+	var websocketHandler controllers.WebsocketHandler
```

Comment:
You're declaring a variable that is interface of type `controllers.WebsocketHandler`, but there is no assignment that implements this interface. Otherwise you're just passing `nil` pointers around. 
You need to declare create a struct that implements `controllers.WebsocketHandler` interface. The struct itself is just a delegate `Upgrade` method to instance of `websocket.Upgrader{}`

```
type DefaultWebsocketHandler struct {
handler websocket.Upgrader
}

func NewWebsocketHandler() WebsocketHandler {
  return &DefaultWebsocketHandler{
    handler: websocket.Upgrader{},
  }
}

func (wh *DefaultWebsocketHandler) Upgrade(w http.ResponseWriter, r *http.Request, responseHeader http.Header) (WebsocketResponseWriter, error) {
     return wh.handler.Upgrade(w, r, responseHeader)
}
```